docs/DeveloperGuide.md

```diff
@@ -1,34 +1,853 @@
+
+1. User input is subjected to `validateInputs()` to verify that 
+both of the following conditions are satisfied.
+  * `weekNumber` is an integer between 1 and 13 inclusive.
+  * `modlist` in `list` is not empty.
+
+If any of the above conditions is not satisfied, an error message will be printed
+and `printAllModuleInformation()` terminates.
+
+1. A method `getMaxModuleLength()` will be called to find out the maximum length 
+of module code of modules taken. This is to facilitate the resizing of the printed table.
+
+1. `updateTemplates()` then updates the table templates such as `border`, `header` 
+and `contents` are updated accordingly based on the output of `getMaxModuleLength()`.
+
+1. For each `Module` in `modList`, `updateContent()` will be called to update the contents
+to be printed.
+
+The following sequence diagram shows how the view module command works. 
+
+![view-module](diagrams/ModViewSequence.jpg)
+
+
+### 4.6 Breakdown and Analysis <a name="4.6"></a>
+
+#### Current Implementation
+
+The analysis command allows the user to view the breakdown of the total time spent
+in the week across all modules and provides a simple analysis of how they are doing.
+
+1. User calls the view module command from the `Parser`. `Parser` then calls the `printBreakdownAnalysis()`
+   method in `Ui`. `printBreakdownAnalysis()` then instantiates `ViewTimeBreakdownAnalysis` class. 
+   and calls the `printTimeBreakdownAndAnalysis()` method. 
+   
+1. The user input is subjected to `validateInputs()` to verify that 
+   both of the following conditions are satisfied.
+     * `weekNumber` is an integer between 1 and 13 inclusive.
+     * `modlist` in `list` is not empty.
+   
+   If any of the above conditions is not satisfied, an error message will be printed
+   and `printTimeBreakdownAndAnalysis()` terminates.
+
+1. `printTime()` method in `printTimeBreakdownAndAnalysis()` is first called. For each `Module` 
+in `modList`, `printTime()` prints `actualWorkload` and `expectedWorkload` in form of a 
+horizontal bar graph. The existence of the desired data is checked using `doesActualTimeExist()` 
+and `doesExpectedTimeExist()` methods of the `Module` class. If the desired data does not exist, a `NO INPUT` will be printed. 
+
+1. `printBreakdown()` then calculates the total time spent by the user on the modules taken in the specified 
+week. It then prints out `actualTime` of each `Module` as a percentage of `totalTimeSpent`. If there is no input, 
+a `NO INPUT` message will be printed. `printBreakdown()` return **TRUE** if there exists a `Module` that has a 
+valid input, else **False** is returned.
+
+1. If output of `printBreakdown()` is `True`, `printAnalysis()` will be called. `computeAnalysisOfTimeSpent()` 
+is called to determine the outcome of the analysis. A message will be printed depending on the output
+of the analysis. 
+
+The following sequence diagram shows how the analysis command works. 
+
+![analysis](diagrams/AnalysisSequence.jpg)
```

Comment:
Should there be return arrows here?

![image](https://user-images.githubusercontent.com/61812576/97656379-4df72a00-1aa2-11eb-995a-e6f7b5a89dfe.png)